src/API/HTTPAdapter/CurlHTTPAdapter.php

```diff
@@ -35,16 +35,21 @@ protected function parseHeaders($headers)
 
     public function request(string $method, string $endpoint, array $params = [], array $headers = []): Response
     {
-        $curl = curl_init(sprintf('%s%s', $this->getBaseUrl(), $endpoint));
-        $headers["Content-Type"] = "application/json";
         $headers["Accept"] = "application/json";
+        $method = strtolower($method);
+
+        if ($method == 'post') {
+            $curl = curl_init(sprintf('%s%s', $this->getBaseUrl(), $endpoint));
+        } else {
+            $curl = curl_init(sprintf( '%s%s?%s', $this->getBaseUrl(), $endpoint, http_build_query($params)));
+        }
```

Comment:
Missing space in 
`$curl = curl_init(sprintf( '%s%s?%s', $this->getBaseUrl(), $endpoint, http_build_query($params)));` 
should be
 `$curl = curl_init(sprintf('%s%s?%s', $this->getBaseUrl(), $endpoint, http_build_query($params)));`

Reply:
https://github.com/NextmediaMa/youcan-payment-php-sdk/pull/8/commits/cd29a3fc2cb76a2fd7606af77fd683ff78e8d7aa